packages/manager/src/features/VPCs/VPCDetail/VPCDetail.tsx

```diff
@@ -166,10 +173,26 @@ const VPCDetail = () => {
           </Typography>
         </Box>
         <Box display="flex" justifyContent="end">
-          <StyledActionButton onClick={() => handleEditVPC(vpc)}>
+          <StyledActionButton
+            disabled={!permissions.update_vpc}
+            onClick={() => handleEditVPC(vpc)}
+            tooltipText={
+              !permissions.update_vpc
+                ? 'You do not have permission to edit this VPC.'
+                : ''
+            }
+          >
             Edit
           </StyledActionButton>
-          <StyledActionButton onClick={() => handleDeleteVPC(vpc)}>
+          <StyledActionButton
+            disabled={!permissions.delete_vpc}
+            onClick={() => handleDeleteVPC(vpc)}
+            tooltipText={
+              !permissions.delete_vpc
+                ? 'You do not have permission to delete this VPC.'
+                : ''
```

Comment:
nit but this is more explicit 👍 (you could choose that pattern accross the PR (I see more instances including above)

```suggestion
              !permissions.delete_vpc
                ? 'You do not have permission to delete this VPC.'
                : undefined
```